app/views/unattended/provisioning_templates/snippet/puppet_setup.erb

```diff
@@ -99,32 +99,14 @@ EOF
 <% end -%>
 <% end -%>
 
-<% if os_family == 'Redhat' -%>
-<% if os_major > 6 -%>
-puppet_unit=puppet
-/usr/bin/systemctl list-unit-files | grep -q puppetagent && puppet_unit=puppetagent
-/usr/bin/systemctl enable ${puppet_unit}
-<% else -%>
-/sbin/chkconfig --level 345 puppet on
-<% end -%>
-<% end -%>
 <% if os_family == 'Freebsd' -%>
 echo 'puppet_enable="YES"' >>/etc/rc.conf
-<% end -%>
-<% unless aio_enabled -%>
-<% if os_family == 'Debian' -%>
-if [ -f "/etc/default/puppet" ]
-then
-/bin/sed -i 's/^START=no/START=yes/' /etc/default/puppet
-fi
-<%= bin_path %>/puppet agent --enable
-<% elsif os_family == 'Suse' -%>
+<% elsif os_family == 'Suse' && !aio_enabled -%>
```

Comment:
I'm not sure there is non-AIO out there for Suse. Perhaps this can just be dropped.